### PR TITLE
Fix -Wstrict-selector-match warning

### PR DESCRIPTION
**Pull request by xinyuxu1026**
## Proposed changes

Office opened [an issue](https://identitydivision.visualstudio.com/Engineering/_boards/board/t/Auth%20Client%20-%20CPP/Backlog%20items/?workitem=2801057) to enable `-Wstrict-selector-match` for obj-c(++) in MSAL, this pr is to fix the warnings.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/claims/MSIDClaimsRequest.m [hunk 1]
```diff
@@ -191,7 +191,7 @@ - (NSDictionary *)jsonDictionary
     
     for (NSNumber *target in self.claimsRequestsDict.allKeys)
     {
-        NSArray *requests = self.claimsRequestsDict[target];
+        NSSet *requests = self.claimsRequestsDict[target];
```
Comment (oldalton) [inline IdentityCore/src/claims/MSIDClaimsRequest.m:194]: Are you sure it's a set? @antrix1989 could you check?

Comment (antrix1989) [reply IdentityCore/src/claims/MSIDClaimsRequest.m:194]: @xinyuxu1026 is right, it should be set here.

File: IdentityCore/src/cache/MSIDMacTokenCache.m [hunk 2]
```diff
@@ -378,15 +378,15 @@ - (BOOL)removeItemsWithKeyImpl:(MSIDCacheKey *)key
         userId = @"";
     }
     
-    NSMutableDictionary *userTokens = [self.cache[@"tokens"] objectForKey:userId];
+    NSMutableDictionary *userTokens = [(NSDictionary *)self.cache[@"tokens"] objectForKey:userId];
```
Comment (antrix1989) [inline IdentityCore/src/cache/MSIDMacTokenCache.m:381]: self.cache is already declared as dictionary:

> @property (nonatomic) NSMutableDictionary *cache;

Why do we need to cast it to NSDictionary here?

Comment (xinyuxu1026) [reply IdentityCore/src/cache/MSIDMacTokenCache.m:381]: The compiler doesn't know the exact type returned by `self.cache[@"tokens"]`, and it's complaining multiple methods named `objectForKey`. And I think it should be `(NSMutableDictionary *)self.cache[@"tokens"]`, there is a check called `validateCache` to make sure Cache always a mutable dictionary in this file.

**Review by antrix1989 — APPROVED** [review]

**Pull request merged**